handlers/deleteitem/deleteItem_test.go

```diff
@@ -0,0 +1,192 @@
+			method:      "GET",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a POST request",
+			path:        "/lists/b6cf642d/items/73bb82c4/",
+			method:      "POST",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a PATCH request",
+			path:        "/lists/b6cf642d/items/73bb82c4/",
+			method:      "PATCH",
+			expectedRes: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dbMocked := &testhelpers.MockDB{}
+			dbMocked.Test(t)
+			defer dbMocked.AssertExpectations(t)
+
+			input := testhelpers.CreateAPIGatewayV2HTTPRequest(tt.path, tt.method, "")
+			d := deleteItem{db: dbMocked}
+			gotRes := d.Match(input)
+
+			assert.Equal(t, tt.expectedRes, gotRes)
+		})
+	}
+}
+
+func TestDeleteItemHandle(t *testing.T) {
+	tests := []struct {
+		name               string
+		path               string
+		listID             string
+		itemID             string
+		output             *data.Item
+		outputErr          error
+		expectedStatusCode int
+		shouldCallDB       bool
+	}{
+		{
+			name:               "Returns 'Bad Request' when the path is empty",
+			path:               "",
+			listID:             "test-list-id",
+			output:             nil,
+			outputErr:          nil,
+			expectedStatusCode: 400,
+			shouldCallDB:       false,
+		},
+		{
+			name:               "Returns 'Bad Request' when the path is not in the correct format",
+			path:               "/lists/test-list-id",
+			listID:             "test-list-id",
+			output:             nil,
+			outputErr:          nil,
+			expectedStatusCode: 400,
+			shouldCallDB:       false,
+		},
+		{
+			name:               "Returns 'OK' and item when the path matches",
+			path:               "/lists/test-list-id/items/test-item-id/",
+			listID:             "test-list-id",
+			itemID:             "test-item-id",
+			output:             &data.Item{Item: "Apples", ID: "888"},
+			outputErr:          nil,
+			expectedStatusCode: 200,
+			shouldCallDB:       true,
+		},
+		{
+			name:               "Returns 'Bad Request' when the item does not exist",
+			path:               "/lists/test-list-id/items/test-item-id/",
+			listID:             "test-list-id",
+			itemID:             "test-item-id",
+			output:             nil,
+			outputErr:          db.NewError(db.ErrorNotFound),
+			expectedStatusCode: 404,
+			shouldCallDB:       true,
+		},
+		{
+			name:               "Returns 'Internal Server Error' when the db returns an error",
+			path:               "/lists/test-list-id/items/test-item-id/",
+			listID:             "test-list-id",
+			itemID:             "test-item-id",
+			output:             nil,
+			outputErr:          errors.New("Something bad happened"),
+			expectedStatusCode: 500,
+			shouldCallDB:       true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dbMocked := &testhelpers.MockDB{}
+			dbMocked.Test(t)
+			if tt.shouldCallDB {
```

Comment:
As before in the update PR about where this should be
https://github.com/mount-joy/thelist-lambda/pull/9/files#r516857435